arch/arm/core/aarch32/cortex_m/fault.c

```diff
@@ -309,8 +318,11 @@ static uint32_t mem_manage_fault(z_arch_esf_t *esf, int from_hard_fault,
 
 				reason = K_ERR_STACK_CHK_FAIL;
 			} else {
-				__ASSERT(0,
+				if (SCB->CFSR &
+					SCB_CFSR_MSTKERR_Msk) {
+					__ASSERT(0,
 					"Stacking error not a stack fail\n");
+				}
```

Comment:
```suggestion
				__ASSERT(!(SCB->CFSR & SCB_CFSR_MSTKERR_Msk),
					"Stacking error not a stack fail\n");
```
Nitpicking: this will save one unnecessary readout of `CFSR` when assertions are not enabled (in addition to reducing the source code by 3 lines).

Reply:
done @anangl thanks